test/evp_extra_test.c

```diff
@@ -1207,6 +1250,10 @@ int setup_tests(void)
     ADD_ALL_TESTS(test_invalide_ec_char2_pub_range_decode,
                   OSSL_NELEM(ec_der_pub_keys));
 #endif
+#ifdef OPENSSL_NO_FIPS
```

Comment:
There is a FIPS_MODE define...

Reply:
> There is a FIPS_MODE define...

Yes, I know. It's this PR that ensures it gets defined! (See providers/build.info)

However that define is intended for files that get compiled twice: once for the default provider, and once for the FIPS provider. Whether it is defined or not doesn't tell you anything about whether FIPS has been disabled for this build via "no-fips". For that we need to use OPENSSL_NO_FIPS - as is done here.

Reply:
What do we do about the unit tests?

I've tried to write them so that the mode doesn't matter when compiled but that's not the same as running them twice, once with and once without, some FIPS mode flag set.  I'd assumed we compile the tests twice.

It would be easy enough to switch over to an environment variable or some other conditional.

Reply:
For unit tests of FIPS code specifically, it would probably be best to have separate internals test programs to run those. They could be built with `FIPS_MODE` always defined and directly using the source they're supposed to test instead of getting it from libcrypto.

As environment variables go, #8663 might provide an answer.